tree_unique_args/src/lib.rs

```diff
@@ -44,9 +50,57 @@ pub enum Expr {
     Let(Id, Box<Expr>, Box<Expr>),
     Arg(Id),
     Function(Id, Box<Expr>),
+    /// A list of functions, with the first
+    /// being the main function.
+    Program(Vec<Expr>),
     Call(Id, Box<Expr>),
 }
 
+impl Expr {
+    pub fn for_each_child(&mut self, mut func: impl FnMut(&mut Expr)) {
```

Comment:
What is `func`?
Could use a doc comment for this function

Reply:
This method runs `func` on each child of `self`
Added a comment